File: app/src/main/java/fr/iutlens/mmi/rogue/Hero.java
package fr.iutlens.mmi.rogue;

import android.graphics.Canvas;

import fr.iutlens.mmi.rogue.util.Coordinate;
import fr.iutlens.mmi.rogue.util.SpriteSheet;

/**
 * Created by dubois on 24/12/2019.
 */

class Hero {
    private final SpriteSheet spriteSheet;
    private final int id;
    private  int x;
    private  int y;

    public Hero(int tileset, int id, int x, int y) {
        this.spriteSheet = SpriteSheet.get(tileset);
        this.id = id;
        this.x = x;
        this.y = y;
    }


    public void paint(Canvas canvas) {
        spriteSheet.paint(canvas,id,x*spriteSheet.w,y*spriteSheet.h,0xffc1b5);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getMobility() {
        return Tile.F_WALK | Tile.F_SWIM;
    }

    public void move(int dir) {
        x += Coordinate.dir_coord[dir][0];
        y += Coordinate.dir_coord[dir][1];
    }
}

File: app/src/main/java/fr/iutlens/mmi/rogue/Hero.kt
package fr.iutlens.mmi.rogue

import android.graphics.Canvas
import android.graphics.Paint
import fr.iutlens.mmi.rogue.util.Coordinate
import fr.iutlens.mmi.rogue.util.SpriteSheet

/**
 * Created by dubois on 24/12/2019.
 */
class Hero(tileset: Int, private val id: Int, var x: Int, var y: Int) {
    private val spriteSheet: SpriteSheet? = SpriteSheet[tileset]

    fun paint(canvas: Canvas) {
        spriteSheet!!.paint(canvas, id, x * spriteSheet.w.toFloat(), y * spriteSheet.h.toFloat(), 0xffc1b5)
    }

    val mobility: Int
        get() = Tile.F_WALK or Tile.F_SWIM

    fun move(dir: Int) {
        x += Coordinate.dir_coord[dir][0]
        y += Coordinate.dir_coord[dir][1]
    }
}